erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -317,7 +318,8 @@ frappe.ui.form.on('Payment Entry', {
 				method: "erpnext.accounts.doctype.payment_entry.payment_entry.get_account_details",
 				args: {
 					"account": account,
-					"date": frm.doc.posting_date
+					"date": frm.doc.posting_date,
+					"cost_center": frm.doc.cost_center
```

Comment:
I don't think party outstanding balance should be fetched based on selected cost center. Party has nothing to do with your cost center. So, mostly they will pay against total outstanding amount.

Reply:
In ideal scenario each branch has its own bank account and maintains branch wise receivable and payable except the case where a company has centralized finance and purchase.
So if company A is having head office in Mumbai and branches in Delhi, Kolkatta and Chennai. Then in this case the branch receivable and payable will settled by branch itself not by Head Office.

Tagging Cost Center in Balance Sheet accounts is similar to maintaining branch wise financial books (e.g.. Trial Balance, P&L and Balance Sheet).

Reply:
I think we would require both pieces of information of cc-wise outstanding and total outstanding. We have cases where customers pay based on total as Navin has commented.

Reply:
Ok, as cost center is optional here, we can fetch outstanding balance as per selected cost center. If user wants to process against total outstanding, then they will leave cost center blank.